project/profiles/capacity4more/modules/c4m/content/c4m_content_photo/c4m_content_photo.module

```diff
@@ -14,16 +14,43 @@ function c4m_content_photo_theme_registry_alter(&$theme_registry) {
 }
 
 /**
+ * Implements hook_node_insert().
+ *
+ * If a photo was uploaded and if it is part of a photo album
+ * rename the file as Album name + number.
+ */
+function c4m_content_photo_node_insert($node) {
+  if ($node->type != 'photo') {
+    return;
+  }
+  if (empty($node->node_gallery_ref_1)) {
+    return;
+  }
+
+  $wrapper = entity_metadata_wrapper('node', $node);
+
+  $source = file_load($wrapper->c4m_media->value()['fid']);
```

Comment:
`$wrapper->c4m_media->value()` - might get it as an array, but it might have the `uri` key.